khanh19934/backend-coding-test
# Implement tooling

## What problem does this PR solve

- Implement tooling for use in application

## How does this PR solve it

- Install `webpack` and `typescript` for bundle and handle typescript file
- Install `eslint` for linting in project
- Install `winston` for logger
- Install `pre hook` for run test before push
- Install `nyc` for check the code coverage
- Setup simple CI with github actions
## How can reviewers test the changes

- Try to run application
- Check the code is changed to Typescript
- Check the setup of `nyc` and the setup for eslint in `.eslintrc`
- Check in terminal to see the logger

## What Documentation/References will help understand the problem and solution

(Please be sure to include links to any libraries, conversations, post, etc...)

## Any limitation or related work yet to be done

- The node version for this project is quite old so it really difficult to setup tooling with latest version. Because many tool is need node version > 10

- I don't have experience with setup CI, so i just follow document of github action for doing that requirement
